src/Functions/assumeNotNull.cpp

```diff
@@ -33,6 +33,7 @@ class FunctionAssumeNotNull : public IFunction
 
     size_t getNumberOfArguments() const override { return 1; }
     bool useDefaultImplementationForNulls() const override { return false; }
+    bool useDefaultImplementationForNothing() const override { return false; }
```

Comment:
Ok, return type for `assumeNotNull(Null)` should be Nothing, and it should throw for > 0 rows

Reply:
Added throwing exception. `useDefaultImplementationForNothing` is not needed here actually, removed